JamesRobertSutcliffe/personality-music-recommender
# Display liked songs in Personalised Homepage

This PR was created so that users have a way of visualising the songs they liked so far, in a similar way to how tracks and albums are displayed. (See [this Discord message](https://discord.com/channels/@me/1002214321532702720/1168255549536735322))

Effort was made to manage liked songs through local state while simultaneously updating the database for future use. This approach ensures users see changes instantly in their current session, with these changes also being preserved for later sessions.

> This one is a big and messy one - it's the result of a feature being developed and continued 3 months later.
> I tried to clean it up the best I could 👀 

Future discussions need to be had about the general User Experience and more specifically about how we want handle Albums.

## Testing
1. Run the server using the script
2. Access your Personalised Homepage
3. Like/Unlike a couple of songs, while confirming the "Liked songs" list is updated in real time
4. Refresh the page to confirm the Liked songs are saved across sessions
